src/BrainPass/BrainPass.sol

```diff
@@ -0,0 +1,343 @@
+        );
+    }
+
+    /// @notice Increase the time to hold a PassNft
+    /// @param tokenId The Id of the NFT whose time is to be increased
+    /// @param newEndTime The new subcription endTime for the of the NFT
+    function increaseEndTime(uint256 tokenId, uint256 newEndTime) external {
+        UserPassItem memory pass = addressToNFTPass[msg.sender][tokenId];
+        if (getUserPassDetails(msg.sender, pass.passId).tokenId != tokenId)
+            revert NotTheOwnerOfThisNft();
+        uint256 newStartTime;
+        if (pass.endTimestamp < block.timestamp) {
+            newStartTime = block.timestamp;
+        } else {
+            newStartTime = pass.endTimestamp;
+        }
+
+        if (!validatePassDuration(newStartTime, newEndTime))
+            revert DurationNotInTimeFrame();
+
+        uint256 price = calculatePrice(pass.passId, newStartTime, newEndTime);
+        bool success = IERC20(iqToken).transferFrom(msg.sender, owner, price);
+        if (!success) revert IncreseTimePaymentFailed();
+
+        UserPassItem memory purchase = UserPassItem(
+            pass.tokenId,
+            pass.passId,
+            pass.startTimestamp,
+            newEndTime
+        );
+
+        addressToNFTPass[msg.sender][tokenId] = purchase;
+        ownerOfToken[pass.passId][msg.sender] = purchase;
+
+        emit TimeIncreased(
+            msg.sender,
+            tokenId,
+            pass.startTimestamp,
+            pass.endTimestamp
+        );
+    }
+
+    /// @notice Withdraws any amount in the contract
+    function withdraw() external payable onlyOwner {
+        uint256 balance = address(this).balance;
+        if (balance <= 0) revert NoEtherLeftToWithdraw();
+        (bool success, ) = (msg.sender).call{value: balance}("");
+        if (!success) revert TransferFailed();
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Internal Functions
+    /// -----------------------------------------------------------------------
+
+    /// @notice Calculate the price of an Nft
+    /// @param startTimestamp The start time to calculate the price of the Nft
+    /// @param endTimestamp The end time to calculate the price of the Nft
+    function calculatePrice(
+        uint256 passId,
+        uint256 startTimestamp,
+        uint256 endTimestamp
+    ) internal view returns (uint256) {
+        PassType memory passType = passTypes[passId];
+        uint256 subscriptionPeriodInSeconds = endTimestamp.sub(startTimestamp);
+
+        uint256 subscriptionPeriodInDays = subscriptionPeriodInSeconds.div(
+            SECONDS_IN_A_DAY
+        );
+
+        // Calculate the total price
+        uint256 totalPrice = subscriptionPeriodInDays.mul(passType.pricePerDay);
+
+        if (passType.discount > 0) {
+            uint256 discountAmount = totalPrice.mul(passType.discount).div(100);
+            totalPrice = totalPrice.sub(discountAmount);
+        }
+
+        return totalPrice * 1e18;
```

Comment:
if you need to multiply price for 1e18 its probably because you are storing it in wrong format.

Each mul in a contract consumes gas. Its much efficient if you get the pricePerDay in the right format that you set once, that doing this each time you need to calculate price